css-plugin-base.js

```diff
@@ -57,14 +57,51 @@ CSSPluginBase.prototype.listAssets = function(loads, opts) {
  * <style> injection browser plugin
  */
 // NB hot reloading support here
+
+function cssInject(style, css) {
+  if (style) {
+      style.innerHTML = css;
+  } else {
+    style = document.createElement('style');
+    style.type = 'text/css';
+    style.innerHTML = css;
+    document.head.appendChild(style);
+  }
+  return style;
+}
+
+// NB the <link> + blob URL trick is required to make chrome detect the source maps
+function cssInjectSourceMaps(link, css) {
+  var href = URL.createObjectURL(new Blob([css], { type:'text/css' }));;
+  if (link) {
+    link.href = href;
+  } else {
+    link = document.createElement('link');
+    link.rel = 'stylesheet';
+    link.href = href;
+    document.head.appendChild(link);
+  }
+  return link;
+}
+
+var elementsContainerKey = '__CSSPluginBase.elements';
 CSSPluginBase.prototype.instantiate = function(load) {
   if (this.builder)
     return;
 
-  var style = document.createElement('style');
-  style.type = 'text/css';
-  style.innerHTML = load.metadata.style;
-  document.head.appendChild(style);
+  var enableInlineSourceMaps = this.inlineCssSourceMaps && load.metadata.styleSourceMap;
+  var cssElements = document[elementsContainerKey] || (document[elementsContainerKey] = {});
```

Comment:
Thanks, 

To fix the hot reloading problem (didn't remove previous style), I reuse the same element if the module was previously loaded.
I store refs to the created elements in a hash ( `{ [address: string]: HTMLStyleElement | HTMLLinkElement }` ) and attach it on the document.
I guess it could also just be a local variable in the module, I was just using document for easier retrieving when debugging.